wrangler-core/src/main/java/io/cdap/directives/column/Swap.java

```diff
@@ -93,4 +99,19 @@ public Mutation lineage() {
       .relation(Many.of(left, right), Many.of(right, left))
       .build();
   }
+
+  @Override
+  public Relation transform(RelationalTranformContext relationalTranformContext,
+                            Relation relation) {
+
+    Optional<ExpressionFactory<String>> expressionFactory = SqlExpressionGenerator
+            .getExpressionFactory(relationalTranformContext);
+    if (!expressionFactory.isPresent()) {
+      return new InvalidRelation("Cannot find an Expression Factory");
+    }
+
+    Relation tempRel =  relation.setColumn("tempColumn", expressionFactory.get().compile(right));
+    tempRel = tempRel.setColumn(right, expressionFactory.get().compile(left));
+    return tempRel.setColumn(left, expressionFactory.get().compile("tempColumn"));
```

Comment:
This adds a new column. Do we need to drop the temp column?

Reply:
Dropping the temp column prevents the swap and returns both columns as is because the query plan somehow doesn't record the creation of the temp column.